File: server/Models/Application.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RedHouse_Server.Models
{
    public class Application
    {
        [Key]
        public int Id { get; set; }
        [ForeignKey(nameof(Property))]
        public int PropertyId { get; set; }
        public Property Property {get; set;}

        [ForeignKey(nameof(User))]
        public int UserId { get; set; }
        public User User { get; set; }
        public DateTime ApplicationDate { get; set; }
        public string Message { get; set; }
        public string ApplicationStatus { get; set; }
        public string ApplicationType { get; set; }
        public double SuggestedPrice { get; set; }

    }
}

File: client/lib/model/application.dart
import 'package:client/model/property.dart';
import 'package:client/model/user.dart';
import 'package:json_annotation/json_annotation.dart';

part "application.g.dart";

@JsonSerializable()
class Application {
  int id;
  int propertyId;
  int userId;
  DateTime applicationDate;
  String applicationStatus;
  String message;
  User user;
  Property property;
  double? suggestedPrice;
  Application({
    required this.id,
    required this.propertyId,
    required this.userId,
    required this.applicationDate,
    required this.applicationStatus,
    required this.message,
    required this.property,
    required this.user,
    this.suggestedPrice,
  });

  factory Application.fromJson(Map<String, dynamic> json) =>
      _$ApplicationFromJson(json);
  Map<String, dynamic> toJson() => _$ApplicationToJson(this);
}
